### PR TITLE
Feature/adding docker comppose

### PR DESCRIPTION
**Pull request by jshilts657**
### Deployment:
- [x] Create Docker compose file and integrate images
- [x] Refactor Docker-compose and code to work in compose structure now. (DB and Backend)
### CI/CD:
- [x] Refactor node portion of pipeline to reduce build times (Down 30-40 secs)
- [x] Test new changes in pipeline.
- [x] Merge James PR and or Dev
### UI:
- [x] Update frontend REST API call to backend
- [x] Resolve styles bug for UI
- [x] Refactor UI to Utilize React design pattern and tools
- [x] Refactor UI Filter fields to reduce query complexity
- [x] Create Query return Display Spot
- [x] Update Unit test so we have some testing
- [x] Create an additional unit test to demonstrate how testing should be done

### Backend:
- [x] Create database Query functions.
- [x] Create New Unit tests
- [x] Update Readme with docker compose directions

### PR:
- [x] Address comments

### CODE REVIEW

**Review by BlueGreenDragon — APPROVED** [review]
Looks good.

**Review by rzheng2019 — APPROVED** [review]
Looks good

**Review by jamzheng109 — APPROVED** [review]
looks good

File: .github/workflows/current-workflow.yml [hunk 1]
```diff
@@ -346,41 +346,31 @@ jobs:
...
       - name: Install Node.js
         uses: actions/setup-node@v4
         with:
           node-version: '22.8.0'
       - name: Install npm
         working-directory: ./code/frontend
-        run: npm ci
-      - name: Install npm react-scripts
-        run: npm install react-scripts --save
+        run: npm ci #&& npm install react-scripts --save
```
Comment (jshilts657) [inline .github/workflows/current-workflow.yml]: clean

File: code/backend/src/backend.py [hunk 2]
```diff
@@ -1,53 +1,87 @@
...
-def get_data():
-    data = request.json
-    input1 = data.get('input1')
-    input2 = data.get('input2')
 
-    # Example logic using the inputs
-    response_message = f"Received input1: {input1}, input2: {input2}"
+# This needs to be refactored to work with container
+def get_db_connection():
+    conn = sqlite3.connect('../../database/src/movies.db')  # Replace with your DB path
```
Comment (jshilts657) [inline code/backend/src/backend.py]: Research how to refactor in docker-compose env

File: code/backend/tests/process_query_request_test.py [hunk 3]
```diff
@@ -9,47 +10,59 @@ def test_client():
...
-                            f"Writer: Monty Python?")
-
-        response_data = json.loads(response.data)
-
-        # Test to ensure the response it correctly formatted
-        assert 'Python Received' in response_data
-        # Verify test has the expected data
-        assert response_data['Python Received'] == expected_message
\ No newline at end of file
+def test_process_data_request(test_client, mocker):
```
Comment (jshilts657) [inline code/backend/tests/process_query_request_test.py:13]: Comment and clean up

File: code/database/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+src/movies.db
```
Comment (jshilts657) [inline code/database/.gitignore:1]: look for cache under src as well and remove

File: code/frontend/src/assets/SubmitFormPage.css [hunk 5]
```diff
@@ -0,0 +1,79 @@
+.container {
```
Comment (jshilts657) [inline code/frontend/src/assets/SubmitFormPage.css:2]: Comment and clean up

File: code/database/.gitignore (hunk 4 above)
Comment (jshilts657) [reply code/database/.gitignore:1]: ignoring for now

**Pull request merged**